contracts/strategies/fluid/FluidStrategy.sol

```diff
@@ -0,0 +1,205 @@
+        uint256 availableLiquidity = _getAvailableLiquidity(_asset);
+        uint256 allocatedValue = allocatedAmount[_asset];
+        if (availableLiquidity > allocatedValue) {
+            interest = availableLiquidity - allocatedValue;
+        }
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function checkLPTokenBalance(address _asset) public view override returns (uint256 balance) {
+        balance = IERC20(_getPTokenFor(_asset)).balanceOf(address(this));
+    }
+
+    /// @notice Internal withdraw function used for withdrawing from the strategy.
+    /// @param _recipient Receiver of the funds.
+    /// @param _asset Asset to be withdrawn.
+    /// @param _amount Amount to be withdrawn.
+    /// @return _amount Amount withdrawn.
+    function _withdraw(address _recipient, address _asset, uint256 _amount) internal returns (uint256) {
+        Helpers._isNonZeroAddr(_recipient);
+        Helpers._isNonZeroAmt(_amount, "Must withdraw something");
+
+        // Checking for shares required to be burned to get the desired _amount and checking maximum redeemable shares.
+        address lpToken = _getPTokenFor(_asset);
+        uint256 shares = IfToken(lpToken).previewWithdraw(_amount);
+        uint256 maxRedeemable = IfToken(lpToken).maxRedeem(address(this));
+        if (shares > maxRedeemable) {
+            revert LimitReached();
+        }
+
+        allocatedAmount[_asset] -= _amount;
+
+        // Redeeming the shares.
+        uint256 received = IfToken(lpToken).redeem(shares, _recipient, address(this));
+        if (received < _amount) {
+            revert Helpers.MinSlippageError(received, _amount);
+        }
+
+        emit Withdrawal(_asset, _amount);
+
+        return _amount;
+    }
+
+    /// @inheritdoc InitializableAbstractStrategy
+    function _abstractSetPToken(address _asset, address _pToken) internal view override {
+        if (IfToken(_pToken).asset() != _asset) {
+            revert InvalidAssetLpPair(_asset, _pToken);
+        }
+    }
+
+    /// @notice A function to fetch the available liquidity deployed in the strategy.
+    /// @param _asset Asset to be checked for available liquidity.
+    /// @return liquidity Available liquidity.
+    function _getAvailableLiquidity(address _asset) internal view returns (uint256 liquidity) {
+        address lpToken = _getPTokenFor(_asset);
+        uint256 lpBalance = IfToken(lpToken).maxRedeem(address(this));
+        liquidity = IfToken(lpToken).convertToAssets(lpBalance);
+    }
+
+    /// @notice Get the lpToken for the asset.
+    ///      Fails if the lpToken doesn't exist in the mapping.
+    /// @param _asset Address of the asset
+    /// @return lpToken to this asset
+    function _getPTokenFor(address _asset) internal view returns (address lpToken) {
+        lpToken = assetToPToken[_asset];
+        if (lpToken == address(0)) revert CollateralNotSupported(_asset);
+        return lpToken;
```

Comment:
```suggestion
```

Reply:
This suggestion is no longer applicable as now the named returns are removed.